python-api-examples/offline-decode-files.py

```diff
@@ -153,6 +194,24 @@ def assert_file_exists(filename: str):
     )
 
 
+def encode_contexts(args, contexts: List[str]) -> List[List[int]]:
+    sp = None
+    if "bpe" in args.modeling_unit:
+        assert_file_exists(args.bpe_model)
+        sp = spm.SentencePieceProcessor()
+        sp.load(args.bpe_model)
+    tokens = {}
+    with open(args.tokens, "r") as f:
```

Comment:
```suggestion
    with open(args.tokens, "r", encoding="utf-8") as f:
```

Reply:
Thanks for the detail explanation, will change to `unique_ptr`, I also find some documents here in case someone else is intrested in it.

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-summary-smartptrs

The items in red boxes are relevant to this comment, other items are also very helpful.

![image](https://github.com/k2-fsa/sherpa-onnx/assets/11765074/7089e497-f55e-4d1f-99c1-fb23e166d142)